listenbrainz/db/feedback.py

```diff
@@ -0,0 +1,152 @@
+import sqlalchemy
+
+from listenbrainz import db
+
+
+def insert(user_id: int, recording_msid: str, score: int):
+    """ Inserts a feedback record for a user's loved/hated recording into the database.
+        If the record is already present for the user, the score is updated to the new
+        value passed.
+        Args:
+            user_id: the row id of the user
+            recording_msid: the MessyBrainz ID of the recording
+            score: The score associated with the recording (+1/-1 for love/hate respectively)
+    """
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO recording_feedback (user_id, recording_msid, score)
+                 VALUES (:user_id, :recording_msid, :score)
+            ON CONFLICT (user_id, recording_msid)
+          DO UPDATE SET score = :score,
+                        created = NOW()
+            """), {
+                'user_id': user_id,
+                'recording_msid': recording_msid,
+                'score': score,
+            }
+        )
+
+
+def delete(user_id: int, recording_msid: str):
+    """ Deletes the feedback record for a given recording for the user from the database.
+        Args:
+            user_id: the row id of the user
+            recording_msid: the MessyBrainz ID of the recording
+    """
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            DELETE FROM recording_feedback
+             WHERE user_id = :user_id
+               AND recording_msid = :recording_msid
+            """), {
+                'user_id': user_id,
+                'recording_msid': recording_msid,
+            }
+        )
+
+
+def get_feedback(user_id: int):
+    """ Get a list of recording feedbacks given by the user.
+        Args:
+            user_id: the row ID of the user in the DB
+        Returns:
+            A list of user's loved and hated recordings
+    """
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+          ORDER BY created DESC
+            """), {
+                'user_id': user_id
+            }
+        )
+        while True:
```

Comment:
this `while true` is a bit scary to me. I know the loop will eventually break, but I'd still prefer if we just used a for loop through all the records, instead of a `while True` and a `break`

Reply:
Removed